src/ol/renderer/canvas/VectorTileLayer.js

```diff
@@ -323,6 +323,8 @@ class CanvasVectorTileLayerRenderer extends CanvasTileLayerRenderer {
     const resolution = frameState.viewState.resolution;
     const rotation = frameState.viewState.rotation;
     hitTolerance = hitTolerance == undefined ? 0 : hitTolerance;
+    const projection = frameState.viewState.projection;
+    coordinate = wrapX(coordinate, projection);
```

Comment:
To avoid in-place changes to the passed `coordinate`, you need to `slice(0)` it. And only call wrapX when the `getWrapX()` from the source returns `true`.